### PR TITLE
[DONE] Various

### PR DESCRIPTION
**Pull request by d-weeteling**
- User can deselect a selected parcel (map zooms to extend of search results)
- User can iterate through results via DetailView
- Omit tablerow seperator in last row in various tables
- Add forgotten _state.ui.selectedParcel_ key to initial state object
- Fix minor layout-related stuff
- Geolocate success results in zooming into your location on the map (Alexander)
- Removed foregroundlayer-chooser from SettingsView (Alexander)
- Login/-out button no longer moves when geolocating
- Show "current index/total" message in DetailView

### CODE REVIEW

File: src/actions/ParcelActions.js [hunk 1]
```diff
@@ -68,15 +73,33 @@ function showSnackBarParcelReceiveError(dispatch, placeName) {
...
+}
+
+export function selectNextParcel(dispatch) {
+  const state = theStore.getState();
+  const oldJsId = state.search.results.indexOf(state.ui.selectedParcel);
+  const newJsId = oldJsId === state.search.results.length - 1 ? 0 : oldJsId + 1;
+  const newParcelId = state.search.results[newJsId];
+  getAttributesFromGeoserver(dispatch, newParcelId);
+  state.ui.selectedParcel = newParcelId;
+}
```
Comment (d-weeteling) [inline src/actions/ParcelActions.js]: refactor ```selectPreviousParcel```

Comment (d-weeteling) [inline src/actions/ParcelActions.js]: done

**Comment by gijs** [comment]
@d-weeteling hij staat nog op 'doing', wil je wel al een code review?

**Comment by d-weeteling** [comment]
Er komen nog wel wat kleine dingen bij, ik laat hem maar even op *doing*
staan.

2017-07-18 8:51 GMT+02:00 Gijs Nijholt <notifications@github.com>:

**Comment by gijs** [comment]
👍 ben bijna op kantoor tot zo

**Comment by d-weeteling** [comment]
Aiht!

2017-07-18 9:03 GMT+02:00 Gijs Nijholt <notifications@github.com>:

File: src/actions/ParcelActions.js [hunk 2]
```diff
@@ -68,15 +73,37 @@ function showSnackBarParcelReceiveError(dispatch, placeName) {
...
 
-export function getAttributesFromGeoserver(dispatch, parcelId) {
-  const currentData = theStore.getState().parcels[parcelId];
-  if (!currentData || !currentData.parcelGeoserverId) {
-    // We can't find the Geoserver featureID
-    return;
+function getNextOrPreviousParcel(dispatch, next = true) {
+  const state = theStore.getState();
+  const oldJsId = state.search.results.indexOf(state.ui.selectedParcel);
+  let newJsId;
```
Comment (gijs) [inline src/actions/ParcelActions.js]: newJsId? `Js` as in Javascript?

File: src/components/DetailViewFooter.jsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
+import PropTypes from "prop-types";
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import { translate, t } from "react-i18next";
+
+// import styles from "./styles/Legend.css";
```
Comment (gijs) [inline src/components/DetailViewFooter.jsx]: rm

File: src/components/DetailViewFooter.jsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+function mapStateToProps(state) {
+  return {
+    searchResults: state.search.results,
+    selectedParcel: state.ui.selectedParcel
+  };
+}
+
+function mapDispatchToProps(dispatch) {
+  return {
+    showPrev: () => selectPreviousParcel(dispatch),
```
Comment (gijs) [inline src/components/DetailViewFooter.jsx]: good stuff - clean API surface

File: src/actions/ParcelActions.js (hunk 2 above)
Comment (d-weeteling) [inline src/actions/ParcelActions.js]: yep, the index in a JS list/"array"

Comment (gijs) [inline src/actions/ParcelActions.js]: 👍

**Pull request merged**